android/pytorch_android/src/main/java/org/pytorch/Tensor.java

```diff
@@ -3,131 +3,167 @@
+    checkShapeAndDataCapacityConsistency(data.length, shape);
+    final DoubleBuffer doubleBuffer = allocateDoubleBuffer((int) numel(shape));
+    doubleBuffer.put(data);
+    return new Tensor_double64(doubleBuffer, shape);
   }
 
-  public static Tensor newFloatTensor(long[] dims, FloatBuffer data) {
+  public static Tensor newTensor(long[] shape, FloatBuffer data) {
     checkArgument(data != null, ERROR_MSG_DATA_BUFFER_NOT_NULL);
-    checkArgument(dims != null, ERROR_MSG_DIMS_NOT_NULL);
-    checkDims(dims);
-    checkDimsAndDataCapacityConsistency(data.capacity(), dims);
+    checkArgument(shape != null, ERROR_MSG_SHAPE_NOT_NULL);
+    checkShape(shape);
+    checkShapeAndDataCapacityConsistency(data.capacity(), shape);
     checkArgument(data.isDirect(), ERROR_MSG_DATA_BUFFER_MUST_BE_DIRECT);
     checkArgument(
         (data.order() == ByteOrder.nativeOrder()),
         ERROR_MSG_DATA_BUFFER_MUST_HAVE_NATIVE_BYTE_ORDER);
-    return new Tensor_float32(data, dims);
+    return new Tensor_float32(data, shape);
   }
 
-  public static Tensor newIntTensor(long[] dims, IntBuffer data) {
+  public static Tensor newTensor(long[] shape, IntBuffer data) {
     checkArgument(data != null, ERROR_MSG_DATA_BUFFER_NOT_NULL);
-    checkArgument(dims != null, ERROR_MSG_DIMS_NOT_NULL);
-    checkDims(dims);
-    checkDimsAndDataCapacityConsistency(data.capacity(), dims);
+    checkArgument(shape != null, ERROR_MSG_SHAPE_NOT_NULL);
+    checkShape(shape);
+    checkShapeAndDataCapacityConsistency(data.capacity(), shape);
     checkArgument(data.isDirect(), ERROR_MSG_DATA_BUFFER_MUST_BE_DIRECT);
     checkArgument(
         (data.order() == ByteOrder.nativeOrder()),
         ERROR_MSG_DATA_BUFFER_MUST_HAVE_NATIVE_BYTE_ORDER);
-    return new Tensor_int32(data, dims);
+    return new Tensor_int32(data, shape);
   }
 
-  public static Tensor newByteTensor(long[] dims, ByteBuffer data) {
+  public static Tensor newTensor(long[] shape, ByteBuffer data) {
     checkArgument(data != null, ERROR_MSG_DATA_BUFFER_NOT_NULL);
-    checkArgument(dims != null, ERROR_MSG_DIMS_NOT_NULL);
-    checkDims(dims);
-    checkDimsAndDataCapacityConsistency(data.capacity(), dims);
+    checkArgument(shape != null, ERROR_MSG_SHAPE_NOT_NULL);
+    checkShape(shape);
+    checkShapeAndDataCapacityConsistency(data.capacity(), shape);
     checkArgument(data.isDirect(), ERROR_MSG_DATA_BUFFER_MUST_BE_DIRECT);
     checkArgument(
         (data.order() == ByteOrder.nativeOrder()),
         ERROR_MSG_DATA_BUFFER_MUST_HAVE_NATIVE_BYTE_ORDER);
-    return new Tensor_byte(data, dims);
+    return new Tensor_byte(data, shape);
   }
 
-  private Tensor(long[] dims) {
-    checkDims(dims);
-    this.dims = Arrays.copyOf(dims, dims.length);
+  private Tensor(long[] shape) {
+    checkShape(shape);
+    this.shape = Arrays.copyOf(shape, shape.length);
   }
 
-  public static long numElements(long[] dims) {
-    checkDims(dims);
+  public static long numel(long[] shape) {
```

Comment:
oh, I meant to add a member method (non-static) that just returns numel of the given tensor. I guess we can have both